injectors/index.js

```diff
@@ -27,34 +27,56 @@ try {
   }
 }
 
-const VALID_PLATFORMS = [ 'stable', 'ptb', 'canary', 'dev', 'development' ];
-
-let platform = (
-  process.argv.find(x => VALID_PLATFORMS.includes(x.toLowerCase())) || 'stable'
-).toLowerCase();
-
-if (!VALID_PLATFORMS.includes(platform)) {
-	console.log(`${AnsiEscapes.RED}Platform you specified isn't valid, please specify a valid one.${AnsiEscapes.RESET}\n\nList of valid platforms:\n${AnsiEscapes.GREEN}${VALID_PLATFORMS.map(x => `${x}`).join('\n')}${AnsiEscapes.RESET}`);
-	process.exit(process.argv.includes('--no-exit-codes') ? 0 : 1);
-}
+let platform = process.argv[4]?.toLowerCase();
 
 (async () => {
-  if (!process.argv.find(x => VALID_PLATFORMS.includes(x.toLowerCase()))) {
-    console.log(`${AnsiEscapes.YELLOW}No valid platform specified, defaulting to "${platform}".${AnsiEscapes.RESET}\n`);
-  } else if (platform === 'development') {
-    platform = 'dev';
+  const exists = main.checkPlatform(platform);
+
+  if (platform) {
+    if (!exists) {
+      console.log(`${AnsiEscapes.RED}Platform you specified isn't valid, please specify a valid one.${AnsiEscapes.RESET}\n\nList of valid platforms:\n${AnsiEscapes.GREEN}${main.VALID_PLATFORMS.map(x => `${x}`).join('\n')}${AnsiEscapes.RESET}`);
+      process.exit(process.argv.includes('--no-exit-codes') ? 0 : 1);
+    } else if (platform === 'development') {
+      platform = 'dev';
+    }
   } else {
-    platform = platform.toLowerCase();
+    platform = 'stable';
```

Comment:
Move code to auto detect platform to here
When checking, don't plug, just check if it's there.